client/control/components/RecipeCombobox.js

```diff
@@ -0,0 +1,181 @@
+import React, { PropTypes as pt } from 'react';
+import cloneArrayValues from 'client/utils/clone-array';
+
+import DropdownMenu from 'control/components/DropdownMenu';
+import GroupMenu from 'control/components/GroupMenu';
+
+/**
+ * Text/dropdown combobox displayed in RecipeFilters.
+ * Used to display and search over filter options
+ * based on user input.
+ */
+export default class RecipeCombobox extends React.Component {
+  static propTypes = {
+    availableFilters: pt.array.isRequired,
+    onFilterSelect: pt.func.isRequired,
+  };
+
+  /**
+   * Utility function to remove a set of properties
+   * from a given object.
+   *
+   * @param  {Object}         object Object to remove props form
+   * @param  {Array<string>}  list   List of properties to remove
+   * @return {Object}         New object without selected properties
+   */
+  static removeProperties(object, list) {
```

Comment:
This should probably be in a util module, since it is generally useful and not specific to this class.